### PR TITLE
Fixed Emanuel's code (B2SHARE + refactoring) and added tasks field to the job type

### PR DESCRIPTION
**Pull request by ubuntolog**
B2SHARE integration has been implemented. Now we can download and save data from S2SHARE in  a volume. Console output from containers will be available to the users

### CODE REVIEW

File: backend-docker/pier/pier.go [hunk 1]
```diff
@@ -58,59 +70,130 @@ func (p *Pier) BuildService(buildDir string) (Service, error) {
...
-		statusMessage = strings.Trim(statusMessage, " ")
-		statusMessage = strings.Replace(statusMessage, "Exited", "Finished", 1)
-		job.Status = statusMessage
+	if !ok {
+		return job, def.Err(nil, "not found")
 	}
+	return job, nil
 }
+
+//
```
Comment (emanueldima) [inline backend-docker/pier/pier.go]: we can remove these lines

File: backend-docker/pier/staging.go [hunk 2]
```diff
@@ -45,34 +46,42 @@ func (p *Pier) DownStreamContainerFile(containerID string, filePath string, w ht
 	return nil
 }
 
-func (p *Pier) StreamVolumeFileList(volumeID string, w http.ResponseWriter) error {
+// ListFiles exported
+func (p *Pier) ListFiles(volumeID VolumeID) ([]VolumeItem, error)  {
 	imageID := "eudatgef/volume-filelist"
```
Comment (emanueldima) [inline backend-docker/pier/staging.go]: where is this image built? we should build it when pier starts?

**Comment by ubuntolog** [comment]
Build fails because docker client cannot execute an image using its tag name, it requires image id. Some changes will be made

File: backend-docker/pier/pier.go [hunk 3]
```diff
@@ -10,6 +10,7 @@ import (
 )
 
 const stagingVolumeName = "Volume Stage In"
+const volumeFileList = "Volume file list"
```
Comment (emanueldima) [inline backend-docker/pier/pier.go]: rename to fileListVolumeName

**Pull request merged**